source/agora/node/Ledger.d

```diff
@@ -305,15 +306,17 @@ public class Ledger
     private void addValidatedBlock (const ref Block block) @safe
     {
         if (!this.storage.saveBlock(block))
-            assert(0);
+            assert(0, format!"Failed to save block: %s"(prettify(block)));
 
         ManagedDatabase.beginBatch();
-        scope (failure) ManagedDatabase.rollback();
+        {
+            scope (failure) ManagedDatabase.rollback();
 
+            this.updateUTXOSet(block);
+            this.updateValidatorSet(block);
+            ManagedDatabase.commitBatch();
+        }
```

Comment:
The extra scope here isn't necessary.

Reply:
Well, I can see why you've done it this way on second look.

But why are logging changes and this change mixed all into the same commit?

Reply:
But without it it can be called from the code that follows. That should not happen as it then asserts and the program stops without logging why.

Reply:
I included it here because this was causing the program to terminate with -6 (IO error) and nothing was printed to help indicate why.

Reply:
Moved changes to #1273